ewatercycle/models/lisflood.py

```diff
@@ -241,6 +241,49 @@ def get_value_as_xarray(self, name: str) -> xr.DataArray:
 
         return da
 
+    def _coords_to_indices(self, name: str, lat: Iterable[float], lon: Iterable[float]) -> Tuple[Iterable[int], Iterable[float], Iterable[float]]:
+        """Convert lat, lon coordinates into model indices."""
+        grid_id = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid_id) # shape returns (len(y), len(x))
+        x_model = self.bmi.get_grid_x(grid_id)
+        y_model = self.bmi.get_grid_y(grid_id)
+        spacing_model = self.bmi.get_grid_spacing(grid_id)
+
+        indices = []
+        lon_converted = []
+        lat_converted = []
+        # in lisflood, x corresponds to lon, and y to lat.
+        # this might not be the case for other models!
+        for x, y in zip(lon, lat):
+            distance, index = find_closest_point(x_model, y_model, x, y)
+            indices.append(index)
+            idy, idx = np.unravel_index(index, shape)
+            lon_converted.append(round(x_model[idx], 4)) # use 4 digits in round
+            lat_converted.append(round(y_model[idy], 4)) # use 4 digits in round
+            # consider a threshold twice of the grid spacing
+            # and convert spacing_model to km using this approximation: 1 degree ~ 111km
+            if distance[idy, idx] > max(spacing_model) * 111 * 2:
+                raise ValueError("This point is outside of the model grid.")
+
+        return np.array(indices), np.array(lon_converted), np.array(lat_converted)
```

Comment:
Adding some whitespace and grouping all the appends can make it more readable:

```suggestion
        for x, y in zip(lon, lat):
            distance, index = find_closest_point(x_model, y_model, x, y)            
            idy, idx = np.unravel_index(index, shape)
            
            # consider a threshold twice of the grid spacing
            # and convert spacing_model to km using this approximation: 1 degree ~ 111km
            if distance[idy, idx] > max(spacing_model) * 111 * 2:
                raise ValueError("This point is outside of the model grid.")
            
            indices.append(index)
            lon_converted.append(round(x_model[idx], 4)) # use 4 digits in round
            lat_converted.append(round(y_model[idy], 4)) # use 4 digits in round

        return np.array(indices), np.array(lon_converted), np.array(lat_converted)
```